core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ProcessService.java

```diff
@@ -56,6 +62,35 @@
     }
     processValidator.validateForCreate(process);
     process.setSpecification(handlerService.handleInsert(process));
+    process.getZones().forEach(zoneKey ->
+      process.getInputs().forEach(input -> consumerService.testCreateConsumer(
+        new Consumer(
+          new ConsumerKey(
```

Comment:
This and the corresponding code in the  update function look like it's duplicated. Can we extract the producer/consumer stuff to their own functions?

Reply:
refactored